runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogData.java

```diff
@@ -242,15 +252,36 @@ void doSerializeInternal(ByteBuf buf) {
             if (data == null) {
                 int lengthIndex = buf.writerIndex();
                 buf.writeInt(0);
-                Serializers.CORFU.serialize(payload.get(), buf);
+                if (hasPayloadCodec()) {
+                    // if the payload has a codec we need to also compress the payload
+                    ByteBuf serializeBuf =  Unpooled.buffer();
+                    Serializers.CORFU.serialize(payload.get(), serializeBuf);
+                    ByteBuffer wrappedByteBuf = ByteBuffer.wrap(serializeBuf.array(), 0, serializeBuf.readableBytes());
+                    ByteBuffer compressedBuf = getPayloadCodecType().getInstance().compress(wrappedByteBuf);
```

Comment:
`compress` takes `ByteBuffer` instead of `ByteBuf`, thus the conversion. Can we expose `compress` instance that takes `ByteBuf` directly within the `Codec` interface?

Reply:
We would need to do the conversion for all implementations of Codec. So I've made doCompressionInternal take Bytebuf directly and centralize the conversion within this method.

Reply:
Can't you have a default implementation in the interface?